spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/cache/CacheEndpoint.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.cache;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.boot.actuate.endpoint.annotation.DeleteOperation;
+import org.springframework.boot.actuate.endpoint.annotation.Endpoint;
+import org.springframework.boot.actuate.endpoint.annotation.ReadOperation;
+import org.springframework.cache.CacheManager;
+import org.springframework.context.ApplicationContext;
+import org.springframework.lang.Nullable;
+
+/**
+ * {@link Endpoint} to expose cache operations.
+ *
+ * @author Johannes Edmeuer
+ * @since 2.0.0
+ */
+@Endpoint(id = "caches")
+public class CacheEndpoint {
+	private final ApplicationContext context;
+
+	public CacheEndpoint(ApplicationContext context) {
+		this.context = context;
+	}
+
+	@ReadOperation
+	public ApplicationCacheManagerBeans cacheManagerBeans() {
+		ApplicationContext target = this.context;
+		Map<String, ContextCacheManagerBeans> contextCacheManagerBeans = new HashMap<>();
+		while (target != null) {
+			Map<String, CacheManagerDescriptor> cacheManagerBeans = new HashMap<>();
+			target.getBeansOfType(CacheManager.class)
+					.forEach((name, cacheManager) -> cacheManagerBeans.put(name,
+							new CacheManagerDescriptor(cacheManager.getCacheNames())));
+			ApplicationContext parent = target.getParent();
+			contextCacheManagerBeans.put(target.getId(), new ContextCacheManagerBeans(
+					cacheManagerBeans, parent == null ? null : parent.getId()));
+			target = parent;
+		}
+		return new ApplicationCacheManagerBeans(contextCacheManagerBeans);
+	}
+
+	@DeleteOperation
+	public void clearCaches(@Nullable String contextId, @Nullable String cacheManagerName,
```

Comment:
I wonder if we need to bring the complexity of `BeansEndpoint` here. If you are working with an application, I don't see the use case of having two cache managers with the same name in different context. 

As a user, having to specify the context is very annoying isn't it?